dusty/commands/utils.py

```diff
@@ -35,9 +38,11 @@ def pty_fork(*args):
     updated_env.update(get_docker_env())
     args += (updated_env,)
     executable = args[0]
+    demote_fn = demote_to_user(get_config_value(constants.CONFIG_MAC_USERNAME_KEY))
 
     child_pid, pty_fd = pty.fork()
     if child_pid == 0:
+        demote_fn()
         os.execle(_executable_path(executable), *args)
```

Comment:
can we not just pass this into a demoted function subprocess and keep all the demoting isolated there?